### DISCUSSION TITLE
GraphQL API - Projects V2 - Create a custom field with options

### DISCUSSION THREAD

**Discussion by jonasclaes** [community/community#35922] [Apps, API and Webhooks]
At the moment the GraphQL API does not support adding new custom fields.

For an application I need to automatically create custom fields. Neither the REST API nor the GraphQL API support this at the moment.

Is it possible to add this?

**Comment by Maximellerbach** [comment]
I would also be interested in this, I would need to assign custom fields to some project cards using the API.
This would be a real plus to automate the creation of issues and directly label them accuratly without having a humain go through the UI.

Thanks !

**Comment by jonasclaes** [comment]
I'm using this for a tool to automatically generate a repository from a name and year. On creation it uses a template repository, creates a team with the necessary team members, and when this feature is complete, I can automatically generate a SCRUM/kanban board for the planning which would make my tool complete and my client happy.

**Reply by Maximellerbach** [reply]
I also want this to automatically create issues and fill them properly to put them in a Kanban.
It seems to be a logical feature, but somehow I can't find it anywhere in the docs :/

**Comment by chicco785** [comment]
I am using this function to create new fields... the issue though is editing them :(

```js
async function createProjectField(
  projectId: string,
  name: string,
  dataType: string,
  token: string,
  singleSelectOptions?: SelectOption[],
): Promise<CreatedProjectField | undefined> {
  try {
  const octokit = new Octokit({ auth: token })
  const response = (await octokit.graphql(
    `
    mutation createProjectField($projectId: ID!, $name: String!, $dataType: ProjectV2CustomFieldType!, $singleSelectOptions: [ProjectV2SingleSelectFieldOptionInput!]) {
      createProjectV2Field(input: { projectId: $projectId, name: $name, dataType: $dataType, singleSelectOptions: $singleSelectOptions }) {
        projectV2Field {
          ... on ProjectV2Field {
            id
            name
          }
          ... on ProjectV2IterationField {
            id
            name
          }
          ... on ProjectV2SingleSelectField {
            id
            name
            options {
              id
              name
            }
          }
        }
      }
    }`,
    { projectId: projectId, name, dataType, singleSelectOptions },
  )) as { createProjectV2Field: { projectV2Field: CreatedProjectField } };

  return response.createProjectV2Field.projectV2Field;
} catch (err) {
  core.error(
    `Failed to create field configuration`
  )
}
return 
};
```

**Reply by fierro** [reply]
@chicco785  did you find a way to edit existing Select fields?

**Reply by chicco785** [reply]
there is no way, the api does not allow for that at the time being...

**Reply by fierro** [reply]
😢 
but thank you for your time @chicco785
